agent/xds/listeners.go

```diff
@@ -851,16 +851,11 @@ func (s *ResourceGenerator) listenersFromSnapshotGateway(cfgSnap *proxycfg.Confi
 				return nil, err
 			}
 		case structs.ServiceKindAPIGateway:
-			// TODO Find a cleaner solution, can't currently pass unexported property types
-			var err error
-			cfgSnap.IngressGateway, err = cfgSnap.APIGateway.ToIngress(cfgSnap.Datacenter)
-			if err != nil {
-				return nil, err
-			}
-			listeners, err := s.makeIngressGatewayListeners(a.Address, cfgSnap)
+			listeners, err := s.makeAPIGatewayListeners(a.Address, cfgSnap)
```

Comment:
This is the crux of the whole change that we're making: instead of converting to an ingress gateway snapshot and generating xDS resources from _that_, we generate xDS resources _directly_ from our API gateway snapshot 🎉